httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/HttpClientBuilder.java

```diff
@@ -751,6 +766,26 @@ public CloseableHttpClient build() {
             }
         }
 
+        if (enableIDNConversion && schemePortResolver == null) {
+            // Load the public suffix list
+            final PublicSuffixMatcher publicSuffixMatcher = PublicSuffixMatcherLoader.getDefault();
+
+            schemePortResolver = new DefaultSchemePortResolver() {
+                @Override
+                public int resolve(final HttpHost host) {
+                    final String hostname = IDN.toASCII(host.getHostName());
```

Comment:
@arturobernalg I am not sure I understand why host name conversion is necessary here. The purpose of the `SchemePortResolver` is to provide the default port for a protocol scheme: `http` -> 80, `https` -> 443, `myhttp` -> 8888. Could you please give an example where the proposed host name translation is necessary?

Reply:
@ok2c 
The purpose of converting the hostname to its ASCII representation using the IDN.toASCII method is to handle domain names that contain non-ASCII characters which may not be supported by some components in the networking stack. By converting the hostname to its ASCII representation, now we ensure that it can be correctly resolved and used by these components.

Reply:
@arturobernalg I understand what the IDN encoding is for. I do not understand what it has to do with the process of the scheme port resolution.

Reply:
@ok2c 
Here's an example of how this conversion could be necessary: let's say you have a URI with an IDN hostname like http://www.öbb.at/, and you are using the SchemePortResolver to determine the default port for the http scheme. If the getHostName method returns the hostname as www.öbb.at, some components in the networking stack may not be able to handle it correctly, and you may get errors or unexpected behavior. By converting the hostname to its ASCII representation (www.xn--bb-eka.at) using IDN.toASCII, you ensure that it can be correctly resolved and used by these components.

Reply:
@arturobernalg Look. I understand what the IDN encoding is for and why it is necessary but please understand what I am trying to say here. The hostname encoding makes no difference on the result of the  `SchemePortResolver`. It produces the same result for both versions because it does not take the host name into consideration. At all. 
```
http://www.öbb.at -> 80
http://www.xn--bb-eka.at -> 80
```
The IDN conversion should take place in `HttpHost` and nowhere else. The `SchemePortResolver` is simply the wrong place.